analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/CaughtException.cs

```diff
@@ -0,0 +1,37 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2022 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.CodeAnalysis;
+using SonarAnalyzer.SymbolicExecution.Constraints;
+using SonarAnalyzer.SymbolicExecution.Roslyn;
+using SonarAnalyzer.SymbolicExecution.Roslyn.OperationProcessors;
+using StyleCop.Analyzers.Lightup;
+
+namespace SonarAnalyzer
+{
+    internal sealed class CaughtException : SimpleProcessor<ICaughtExceptionOperationWrapper>
+    {
+        protected override ICaughtExceptionOperationWrapper Convert(IOperation operation) =>
+            ICaughtExceptionOperationWrapper.FromOperation(operation);
+
+        protected override ProgramState Process(SymbolicContext context, ICaughtExceptionOperationWrapper operation) =>
+            context.SetOperationConstraint(ObjectConstraint.NotNull);
+    }
```

Comment:
Operation is not used => we can simplify
```suggestion
internal sealed class CaughtException : ISimpleProcessor
{
    public ProgramState Process(SymbolicContext context) =>
        context.SetOperationConstraint(ObjectConstraint.NotNull);
}
```

Reply:
Looking at this, it's complete duplication with `Creation` class. With a little bit of imagination, caught exception operation creates the instance, so I'd remove this file and just reuse the Creation for this operation type too

Reply:
I wouldn't do that. Creation shouldn't set the "NotNull" constraint unconditionally (e.g. any value types or type parameters constrained to structs).